tests/cpp/test_scatter.cpp

```diff
@@ -209,7 +271,9 @@ TEST_F(ScatterTest, BlockCountingWithShmem2D) {
   tv4->setAllocationDomain(tv4->getLogicalDomain(), true);
 
   auto options = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
-  auto t0 = at::randperm(self_shape[1], options).slice(0, 0, index_shape[1]);
+  auto t0 = at::randperm(self_shape[1], options)
+                .slice(0, 0, index_shape[1])
+                .repeat({index_shape[0], 1});
```

Comment:
~How did it work before?! without the repeat, the input shape mismatches the definition... I'm surprised we haven't hit any error before.~

NVM, this test is still being skipped.

Reply:
another nitpick. I think you would want to slice it with the proper shape (or better yet, just call as_strided).
Because with `repeat`, you have non-unique index, which requires accumulate on the `scatter` call, without being semantically wrong.